src/main/java/com/sap/cloudfoundry/client/facade/rest/CloudControllerRestClientImpl.java

```diff
@@ -921,17 +929,47 @@ private CloudServiceInstance getServiceInstanceIfRequired(String serviceInstance
         return serviceInstance;
     }
 
+    @Override
+    public CloudServiceBinding getServiceBinding(UUID serviceBindingGuid) {
+        // TODO: Refactor and use serviceBindingsV3.get() when https://github.com/cloudfoundry/cf-java-client/pull/1171 is merged and
+        // released
+        return fetch(() -> getServiceBindingResource(serviceBindingGuid), ImmutableRawCloudServiceBinding::of);
+    }
+
+    private Mono<ServiceBindingResource> getServiceBindingResource(UUID serviceBindingGuid) {
+        IntFunction<ListServiceBindingsRequest> pageRequestSupplier = page -> ListServiceBindingsRequest.builder()
+                                                                                                        .page(page)
+                                                                                                        .id(serviceBindingGuid.toString())
+                                                                                                        .build();
+        return PaginationUtils.requestClientV3Resources(page -> delegate.serviceBindingsV3()
+                                                                        .list(pageRequestSupplier.apply(page)))
+                              .singleOrEmpty();
+    }
+
     @Override
     public List<CloudServiceBinding> getServiceAppBindings(UUID serviceInstanceGuid) {
         return fetchList(() -> getServiceBindingResourcesByServiceInstanceGuid(serviceInstanceGuid), ImmutableRawCloudServiceBinding::of);
     }
 
+    @Override
+    public List<CloudServiceBinding> getAppBindings(String applicationName) {
+        UUID applicationGuid = getRequiredApplicationGuid(applicationName);
+        return fetchList(() -> getServiceBindingResourcesByApplicationGuid(applicationGuid), ImmutableRawCloudServiceBinding::of);
+    }
+
     @Override
     public CloudServiceBinding getServiceBindingForApplication(UUID applicationId, UUID serviceInstanceGuid) {
         return fetch(() -> getServiceBindingResourceByApplicationGuidAndServiceInstanceGuid(applicationId, serviceInstanceGuid),
                      ImmutableRawCloudServiceBinding::of);
     }
 
+    @Override
+    public CloudServiceBinding getServiceBindingForApplication(String applicationName, String serviceInstanceName) {
```

Comment:
Is it really necessary? Can we just get app and service guids separately and pass them to the already existing method?

Reply:
Removed